### ISSUE TITLE
Question: Simplify order of terms for `5i+4`

### ISSUE THREAD

**Issue by gwhitney** [josdejong/mathjs#2407]
At the moment, `simplify()` prefers adding "constants" last, and the rules that implement that make no distinction according to identity of the variable(s). In case "i" happens to be a "regular" variable, like an integer index, the order that simplify produces -- `5i+4` -- would be the conventionally preferred one. But if `i` is being interpreted as the imaginary unit, then of course `4+5i` would be the typical notation. So two questions:

1) In the vanilla case `simplify('5i+4')` with no other information, which interpretation/assumptions would you want to make? Should this default output be changed  to `4+5i` ?
2) Should this be configurable? If so, I presume with an options flag, something like `{ imaginaryi: true }` (with the default for the flag corresponding to the answer to Q1).

**Comment by josdejong** [comment]
That is a really good point, `simplify` simply has no idea about the meaning of `i`. I think this falls in the same category as what we discuss in #927: simplifying `sin(pi)`, `log(e)`, `phi^2`, etc based on the meaning of the function. 

To me it makes sense that by default, `simplify` handles `i` as the imaginary number, and it makes sense to be able to turn this off. I think the same holds for treating other functions like `sin`, `log`, etc. Maybe we can implement a configurable map with function and variable names and how/if they are treated as a known list of mathematical functions by `simplify`. 

For reference: #927

**Comment by gwhitney** [comment]
OK, sounds good. Once simplifyCore has settled into its new place from #2405, I will work on a PR for this (since presumably something like `simplifyCore('2i+4+3i')` should by default produce `4+5i` by analogy with `simplifyCore('2+3')`, so this will need to modify simplifyCore. One proposal for controlling this: use the context, now that's an argument to simplify -- check on whether `i` has the property `asBuiltin`, which it would by default. Then it could be turned off by an options argument `{context: {i: {asBuiltin: false}}}`.

**Comment by gwhitney** [comment]
(An additional point of the proposal for controlling this is that it would seem to naturally extend to `sin`, `log`, etc., by marking them in the context as `asBuiltin: false`.)

**Comment by josdejong** [comment]
> One proposal for controlling this: use the context, now that's an argument to simplify -- check on whether `i` has the property `asBuiltin`, which it would by default. Then it could be turned off by an options argument `{context: {i: {asBuiltin: false}}}`.

Yes that makes sense 👍. And indeed that same mechanism could work for other functions like `sin`, `log`. 

I'm trying to come up with a better name for `asBuiltin`, it sounds a big vague to me. Something like `withSematicMeaning` or so? (though this is quite long).

**Comment by gwhitney** [comment]
`standard`? `conventional`? `standardDefinition`? Just brainstorming.
